From f1b71bc3ff09161183d7ef24aadc6b169d903142 Mon Sep 17 00:00:00 2001
From: Hisashi Nakamura <hisashi.nakamura.ak@renesas.com>
Date: Tue, 10 Apr 2018 08:28:07 +0700
Subject: [PATCH 118/433] spi: rspi: Fix SSLKP last SPCMD0

Reference path:
- git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-backport.git
- commit: <a475ddac3fd619ff0741186acfb819bba0fc4376>

    If SSLKP(SSL Signal Level Keeping) is 1, chip select will not be
    negated until SPCR.CPE is set to 0. SSLKP is set to 0 to last SPCMD.

Signed-off-by: Hisashi Nakamura <hisashi.nakamura.ak@renesas.com>
Signed-off-by: khanhluu <khanh.luu.xw@rvc.renesas.com>
Signed-off-by: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
Signed-off-by: Loc Vu <loc.vu.zn@renesas.com>
---
 drivers/spi/spi-rspi.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/spi/spi-rspi.c b/drivers/spi/spi-rspi.c
index d642891..8ebd2e2 100644
--- a/drivers/spi/spi-rspi.c
+++ b/drivers/spi/spi-rspi.c
@@ -987,6 +987,10 @@ static int qspi_setup_sequencer(struct rspi_data *rspi,
 	}
 	if (i) {
 		/* Set final transfer data length and sequence length */
+		rspi_write16(rspi,
+			rspi_read16(rspi, RSPI_SPCMD(i - 1)) &
+			~SPCMD_SSLKP,
+			RSPI_SPCMD(i - 1));
 		rspi_write32(rspi, len, QSPI_SPBMUL(i - 1));
 		rspi_write8(rspi, i - 1, RSPI_SPSCR);
 	}
-- 
2.7.4

